pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -252,13 +252,16 @@ func (m *managerImpl) synchronize(diskInfoProvider DiskInfoProvider, podFunc Act
 	// build the ranking functions (if not yet known)
 	// TODO: have a function in cadvisor that lets us know if global housekeeping has completed
 	if m.dedicatedImageFs == nil {
-		hasImageFs, splitDiskError := diskInfoProvider.HasDedicatedImageFs(ctx)
-		if splitDiskError != nil {
-			klog.ErrorS(splitDiskError, "Eviction manager: failed to get HasDedicatedImageFs")
-			return nil, fmt.Errorf("eviction manager: failed to get HasDedicatedImageFs: %w", splitDiskError)
+		hasImageFs, imageFsErr := diskInfoProvider.HasDedicatedImageFs(ctx)
+		if imageFsErr != nil {
+			klog.ErrorS(imageFsErr, "Eviction manager: failed to get HasDedicatedImageFs")
+			return nil, fmt.Errorf("eviction manager: failed to get HasDedicatedImageFs: %w", imageFsErr)
 		}
 		m.dedicatedImageFs = &hasImageFs
-		splitContainerImageFs := m.containerGC.IsContainerFsSeparateFromImageFs(ctx)
+		splitContainerImageFs, splitErr := diskInfoProvider.HasDedicatedContainerFs(ctx)
+		if splitErr != nil {
+			klog.ErrorS(splitErr, "eviction manager: failed to check if we have separate container filesystem. Ignoring.")
```

Comment:
What's the impact of ignoring the error and assuming container and image filesystem are shared? 
(in line 258 we error out when "HasDedicatedImageFs" returns an error)

Reply:
The first try of this PR returned an error if we couldn't check split fs. This lead to us not setting any eviction signals so we were hitting disk pressure limits and we would never trigger any eviction signals.

We set the signals only once and we re use those. Since we return once, we skip setting signals.

In case of container and image fs are shared, we don't really even need the containerfs signal.

Reply:
So if there's an error, we assume shared container and image filesystem. This is not great but at least we'll continue the evaluation and try to evict the pods if necessary. Is that right? 

With the same reasoning, why shouldn't we ignore the `imageFsErr`? Just trying to understand if I missed something.

Reply:
I thought about it but that has been that way forever. So I thought against changing that.

Reply:
And yes, your thought process is correct.

Reply:
I guess I'm not sure in what situations we'd have permanent errors vs transient ones. If errors are usually transient, returning it and retry next time seems reasonable. If they could be permanent, continue and do best-effort eviction seems more desirable. In that case, maybe kubelet should send an event to surface the issue. 

Have we seen any errors being reported in real life? 

We could at least document why we ignore the error and add a TODO to revisit the error handling of both later.

Reply:
I added a comment. This PR wasn't really about fixing the imagefs path so I left it alone for now.